lib/pigeon/apns/config_parser.ex

```diff
@@ -45,15 +45,13 @@ defmodule Pigeon.APNS.ConfigParser do
   def file_path(nil), do: nil
 
   def file_path(path) when is_binary(path) do
-    path
-    |> Path.expand()
-    |> validate_file_path!()
+    if :filelib.is_file(path), do: Path.expand(path), else: nil
```

Comment:
It might be better to return `{:error, path}` so that the path is not lost during debugging.

Reply:
The big issue is not knowing whether a raw body or file path is given to `:cert` and `:key`. It attempts both file lookup and PEM decode logic on the given value, then sets both on the underlying struct. 

If a raw body is given, `:certfile` and `:keyfile` would have an `{:error, value}` set, which will also need to be redacted. Might be easier to just set it to `{:error, :nofile}` instead.

Reply:
Wouldn’t you know by time you hit `validate!` because you attempted a PEM decode on the value provided? Ultimately, it’s not a big deal to me, as I’m only ever going to be providing a raw body to either (either from the database or some other parameter store like Hashicorp Vault). As long as we properly obfuscate secrets, I’m good. For those people who put pathnames in place…I am thinking that the filename would be _useful_, but you’d have to basically try to PEM decode the filename before writing `{:error, value}`. It might be useful to have the logic be something like:

```elixir
case decode_pem(value) do
  {:ok, value} ->
    {:cert, value}
  :error ->
    if(is_file(value), do: {:certfile, value}, else: {:error, {:nofile, value}})
end
```

Just a thought. You were thinking of dropping the automatic application and changing to a fully (manually) supervised model as a breaking change: it may be a good opportunity to break this as well and only accept PEM data, forcing clients to do the `File.read!` on their filenames.

Reply:
I think in general if an operation fails, it shouldn't return nil. The correct behavior should return either return a tuple of the form `{:error, _}` (when the user is expected to handle the result), or raise an exception.

In its current form `file_path/1` returns `nil` both when a path wasn't given (hence the argument nil), but also when a path was given, but it is invalid/non-existing.